craft_application/services/testing.py

```diff
@@ -105,8 +108,27 @@ def run_spread(self, spread_dir: pathlib.Path) -> None:
                 retcode=exc.returncode,
             )
 
+    def _get_backend_type(self) -> str:
+        return "ci" if os.environ.get("CI") else "lxd-vm"
+
+    def _get_system(self) -> str:
+        name = self._get_backend_type()
+
+        if name == "ci":
+            try:
+                osrel = os_release.OsRelease()
+                os_id = osrel.id()
+                version_id = osrel.version_id()
+                system = f"{os_id}-{version_id}"
```

Comment:
```suggestion
                distro_base = craft_platforms.DistroBase.from_linux_distribution(distro.LinuxDistribution())
                system = f"{distro_base.distribution}-{distro_base.series}"
```

Reply:
Refactored to use this. Snapcraft could also use it replacing `os_release`.